### DISCUSSION TITLE
Change environment variable names

### DISCUSSION THREAD

**Discussion by shrynx** [adonisjs/core#4673] [Help]
Hi !

I am trying to use adonis for my project. we have a mono repo for our frontends and want to start using adonis for our backend.
since it is a mono repo we have single .env file at root shared between all projects we have marked port for each project with their name 
ADMIN_UI_PORT etc. and want to do same for backend  API_PORT, but i can't map this anywhere in adonis js.

**Comment by shrynx** [comment]
Checking the internal whay i would suggest in having a env variables object passed around. this will come from mapping of process.env after being validated by the adonisjs/env package and they keys of the object can the one in the internal representation

**Reply by thetutlage** [reply]
Maybe you can update the `bin/server.ts` file and assign the `PORT` variable from the `API_PORT` variable.

```ts
process.env.PORT = process.env.API_PORT
```

**Reply by shrynx** [reply]
yeah that's the hack i currently settled for.

**Reply by thetutlage** [reply]
Wouldn't call it a hack. `PORT` and `HOST` are the two environment variables which are referenced in a hardcoded manner in the AdonisJS codebase. Since, in 90% of the use-cases these environment variables are provided (and also a standard), we think adding the ability to tweak them is unnecessary for now.

And this is where the ability to self modify the `bin/server.ts` file comes into the picture. We keep all the boot files (the one's inside the `bin` folder) as part of your project, so that you can modify them as needed.
